hack/verify-golangci-lint.sh

```diff
@@ -18,18 +18,27 @@ set -o errexit
 set -o nounset
 set -o pipefail
 
-VERSION=v1.42.0
+VERSION=v1.43.0
 URL_BASE=https://raw.githubusercontent.com/golangci/golangci-lint
 URL=$URL_BASE/$VERSION/install.sh
+INSTALL_CHECKSUM=294771225087ee48c8e0a45a99ac82ed8f9c6e9d384e692ab201986479c8594f
 
 if [[ ! -f .golangci.yml ]]; then
     echo 'ERROR: missing .golangci.yml in repo root' >&2
     exit 1
 fi
 
 if ! command -v golangci-lint; then
-    curl -sfL $URL | sh -s $VERSION
-    PATH=$PATH:bin
+    INSTALL_SCRIPT=$(mktemp -d)/install.sh
+    curl -sfL $URL > $INSTALL_SCRIPT
+    if echo "${INSTALL_CHECKSUM} $INSTALL_SCRIPT" | sha256sum --check; then
+        chmod 755 $INSTALL_SCRIPT
+        $INSTALL_SCRIPT $VERSION
+        PATH=$PATH:bin
+    else
+        echo 'ERROR: failed to install golangci-lint' >&2
```

Comment:
Let's output here the actual reason of why the install failed:
```suggestion
        echo 'ERROR: install script sha256 checksum invalid' >&2
```

Reply:
Done.